packages/synapse-react-client/src/mocks/msw/handlers/tableQueryHandlers.ts

```diff
@@ -54,85 +55,44 @@ export function getHandlersForTableQuery(
   response: QueryResultBundle,
   backendOrigin = getEndpoint(BackendDestinationEnum.REPO_ENDPOINT),
 ) {
-  // We must map generated async job IDs to requests to disambiguate multiple calls to these endpoints
-  const mapOfRequests = new Map<string, QueryBundleRequest>()
+  return generateAsyncJobHandlers<QueryBundleRequest, QueryResultBundle>(
+    TABLE_QUERY_ASYNC_START(':id'),
+    tokenParam => TABLE_QUERY_ASYNC_GET(':entityId', tokenParam),
+    request => removeBundleFieldsUsingMask(response, request.partMask),
+    backendOrigin,
+  )
+}
 
-  return [
-    rest.post(
-      `${backendOrigin}${TABLE_QUERY_ASYNC_START(':id')}`,
-      async (req, res, ctx) => {
-        const asyncJobId = uniqueId()
-        mapOfRequests.set(asyncJobId, await req.json())
-        return res(
-          ctx.status(201),
-          ctx.json<AsyncJobId>({
-            token: asyncJobId,
-          }),
-        )
-      },
-    ),
+export function getAnnotationColumnHandlers(
+  response: ViewColumnModelResponse,
+  backendOrigin = getEndpoint(BackendDestinationEnum.REPO_ENDPOINT),
+) {
+  return generateAsyncJobHandlers<
+    ViewColumnModelRequest,
+    ViewColumnModelResponse
+  >(
+    '/repo/v1/column/view/scope/async/start',
+    tokenParam => `/repo/v1/column/view/scope/async/get/${tokenParam}`,
+    response,
+    backendOrigin,
+  )
+}
```

Comment:
Creating new mock services that use async jobs is now super easy too

Reply:
Would it be helpful (for project spinup) document common tasks and solutions like this (perhaps links in the README?)?

Reply:
Totally agree. Wondering how much effort to put into that considering the following:
1. MSW just released a new major version that (I think) simplifies writing these handlers
2. What we have today could use a bit of cleaning up before we try to 'standardize' it
3. I'm also kind of hoping we can find a tool that makes it easier to generate these stubs using the OpenAPI spec when that work is done